clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -50,15 +60,30 @@ private ClientUtils() {}
                     if (host == null || port == null)
                         throw new ConfigException("Invalid url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
 
-                    InetSocketAddress address = new InetSocketAddress(host, port);
-
-                    if (address.isUnresolved()) {
-                        log.warn("Removing server {} from {} as DNS resolution failed for {}", url, CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, host);
+                    if (clientDnsLookupBehaviour == ClientDnsLookup.RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY) {
```

Comment:
can we also check if the security.protocol is sasl or sasl_ssl and only enable this for the sasl handshake. Instead of asking users to disable this if they are using SSL?

Reply:
Hi,

This has been suggested previously and was part of the initial implementation (see https://github.com/apache/kafka/pull/4485/commits/24757eb7b06bcf8c7d7649c85232c52b5d54f0e4)

You can potentially use SASL_SSL and still enable this feature.
This has been discussed on the email thread (see replies https://lists.apache.org/thread.html/ec3b6826f876382690214cdcad3ab9adc8f0986b34feb799c9810f22@%3Cdev.kafka.apache.org%3E)